packages/document/module-doc/docs/en/api/config/build-config.mdx

```diff
@@ -647,6 +644,44 @@ export default defineConfig({
 });
 ```
 
+## hooks
+
+Build lifecycle hooks that allow custom logic to be injected at specific stages of the build process.
+
+- **Types**:
+
+```ts
+type HookList = {
+  name: string;
+  apply: (compiler: ICompiler) => void.
+}
+
+```
+
+- **Default**: `[]`.
+
+We can get the compiler instance in the apply method, modify its properties, and execute custom logic at different stages. For more information on Hooks, see [Using Hooks to Intervene in the Build Process](see [Using Hooks to Intervene in the Build Process]).
+For more information on Hooks, see [Using-hooks-to-intervene-in-the-build-process](/guide/advance/in-depth-about-build.html#using-hooks-to-intervene-in-the-build-process).
+
+```ts title="modern.config.ts"
+export default defineConfig({
+  buildConfig: {
+    buildType: 'bundle',
+    hooks: [
```

Comment:
I think this is kind of high level usage and should be used by advanced users.

Should this be a plugin api instead of a build config?

Reply:
The plugin API works on every object in the buildConfig, and it wasn't granular enough that I added hooks to the configuration

Reply:
This "hook" is actually a "plugin" for single build config 😂

Can we enhance the plugin API to allow it apply to single build config?

Reply:
Not good unless you change the plugin API

Reply:
This means that, for example, the babel plugin and the changeset plugin are fundamentally different.